samcli/local/docker/manager.py

```diff
@@ -64,13 +125,20 @@ def run(self, container, input_data=None, warm=False):
         else:
             LOG.info("Requested to skip pulling images ...\n")
 
-        if not container.is_created():
-            # Create the container first before running.
-            # Create the container in appropriate Docker network
-            container.network_id = self.docker_network_id
-            container.create()
-
-        container.start(input_data=input_data)
+        try:
+            if not container.is_created():
+                # Create the container first before running.
+                # Create the container in appropriate Docker network
+                container.network_id = self.docker_network_id
+                container.create()
+
+            container.start(input_data=input_data)
+        except docker.errors.APIError as api_error:
+            if api_error.status_code == 409:
+                raise DockerContainerException("'{}' Docker container name is already taken".format(container.name))
+            if api_error.is_server_error():
+                raise DockerContainerException("Something went wrong on the Docker server")
```

Comment:
Instead of this custom error message, I would pass the actual exception message returned by Docker API - `str(ex)`. This will add more information.

Reply:
Maybe let's LOG.debug the actual exception? I don't think custom needs to know the exact bad thing, that happened, or he does?